lib/Driver/Compilation.cpp

```diff
@@ -804,6 +809,9 @@ namespace driver {
           // To avoid redoing the batch loop too many times, repartition pretty
           // aggressively by doubling partition count / halving size.
           NumPartitions *= 2;
+#ifndef NDEBUG
+          llvm::errs() << "Should have used a supplementary output file map.\n";
```

Comment:
I don't think we should be writing random things to `llvm::errs()` without a specific flag. `llvm::dbgs()` might be a good alternative (requires a flag to turn *on).*

Reply:
I was wondering about that. Changed to dbgs. Keep the NDEBUG conditional?

Reply:
Ah, I think using the LLVM `DEBUG` macro is sufficient, no need for `NDEBUG`.

Reply:
Cool! Adding a DEBUG_TYPE called BATCH_MODE, hope that's right.